src/ToricVarieties/LineBundles/methods.jl

```diff
@@ -0,0 +1,115 @@
+############################
+# Line bundle cohomologies
+############################
+
+@doc Markdown.doc"""
+    all_cohomologies(l::ToricLineBundle)
+
+Computes the dimension of all sheaf cohomologies of the toric line bundle `l`.
+
+# Examples
+```jldoctest
+julia> dP3 = del_pezzo(3)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> all_cohomologies(ToricLineBundle(dP3, [1,2,3,4]))
+3-element Vector{fmpz}:
+ 11
+ 0
+ 0
+```
+"""
+function all_cohomologies(l::ToricLineBundle)
+    # check if we can apply cohomCalg
+    v = variety(l)
+    if !((issmooth(v) && iscomplete(v)) || (issimplicial(v) && isprojective(v)))
+        throw(ArgumentError("cohomCalg only applies to toric varieties that are either smooth, complete or simplicial, projective."))
+    end
+    
+    # extract vector of currently-known cohomology dimensions (or create it if necessary)
+    return get_attribute!(l, :all_cohomologies) do
+        
+        # Minimal example:
+        #
+        # The following lines provide an interface to cohomCalg. This is easiest explained with an example.
+        # 
+        # Example: Compute the cohomologies of the structure sheaf of the 2-dimensional projective space.
+        #
+        # Step 1: We trigger the following command so that cohomCalg computes these cohomologies for us:
+        #
+        # ./cohomCalg "vertex x1 | GLSM: ( 1 ); vertex x2 | GLSM: ( 1 ); vertex x3 | GLSM: ( 1 ); srideal [x1*x2*x3]; ambientcohom O(0 );"
+        #
+        # The notation GLSM is for generalized linear sigma model.
+        # This is a physics model which famously triggered interest in toric geometry for physics applications.
+        # The cohomCalg algorithm relies only on the grading of the Cox ring and the Stanley-Reisner ideal.
+        # <-> Those two pieces of information are (more or less) the defining data of a GLSM.
+        #
+        # Step 2: We receive the following return value from cohomCalg:
+        #
+        # "{True,{{1,0,0},{{0,1*1}}}}"
```

Comment:
What does the `1*1` mean?

Reply:
Simple question - complicated answer. This is why I did not elaborate on this in my original explanation, other than saying that it encodes "intermediate results". Still, since you are asking, I have just extended this explanation:

        # By the cohomCalg algorithm, the line bundle cohomologies are given by certain rationoms, that is fractions of 
        # monoms in the Cox ring of the variety. Behind the scenes, cohomCalg reads-off monomials which contribute
        # to the denominators of these rationoms. For sake of simplicity, let us refer to these monomials as "partial-denominators".
        # 
        # Crucially: These partial-denominators are read-off from the Stanley-Reisner ideal. Hence, they encode information about
        # the toric space and are not specific to a certain line bundle. In https://arxiv.org/abs/1802.08860, this is used to infer
        # refined vanishing sets of line bundle cohomologies on toric spaces. Implementations of which are currently available in
        # https://github.com/homalg-project/ToricVarieties_project and will be migrated to OSCAR soon.
        # 
        # To make contact with a line bundle in question, pick a partial-denominator and look for all rationoms with multi-degree
        # matching that of the line bundle in question. In particular, under the assumptions that the variety in question be either
        # smooth, complete or alternatively simplicial, projective, one finds a finite number of such rationoms for each partial-denominator.
        # 
        # Once these rationoms are identified, two questions remain:
        # (1) To which line bundle cohomology do they contribute?
        # (2) Could they encode more than a one-dimensional linear subspace of these cohomology spaces?
        # 
        # The answers to both questions are involved and described in detail in the original work https://arxiv.org/pdf/1003.5217.pdf
        # and the proofs of this algorithm in https://arxiv.org/abs/1006.2392, https://arxiv.org/abs/1006.0780. In particular note that
        # the answer to (2) involved the evaluation of certain "remnant cohomologies", so that a single rationom may indeed encode
        # a vector space of line bundle cohomologies which is strictly larger than 1.
        # 
        # Let us now return to the above return value {{0,1*1}}. The fact that this list contains only a single sub-list states that only 
        # a single partial-denominator was identified. The first value tells to which cohomology group it contributes. Here we find 0,
        # so it contributes to H^0. The second value is, in the following order, the product of the multiplicity (as in response to 
        # (2) above) and the exact form of the partial-denominator. So here the multiplicity is 1 and the partial-denominator is 1.
        # For other examples, we could for example find "2 * x2 * x3 * x5", which would mean that the multiplicity is 2
        # and the partial denominator in question is "x2 * x3 * x5".